### ISSUE TITLE
Precedence of subscripts and superscripts

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#3198]
There's a comment in `HyperplaneArrangements` which I think we should fix by changing the precedence of _ and ^:
```m2
Flat ^ Flat := Flat => meet  -- ooh, cool.  But note L_1^L_2 isn't L_1^(L_2) !
```

--
Initially this issue was the following, but I switched it because I think the above is much more important.
> `join` for posets is called `posetJoin`, but would be more intuitive if `join` was available as  a method. Similarly, flats in `HyperplaneArrangements` have `meet` (or `^`) and `vee` (or `|`) rather than `join`.

**Comment by mahrud** [comment]
Similarly, flats in `HyperplaneArrangements` have `meet` (or `^`) and `vee` (or `|`) rather than `join`.

Also, there's a comment there which I think we could fix by changing the precedence, if it doesn't break anything else:
```m2
Flat ^ Flat := Flat => meet  -- ooh, cool.  But note L_1^L_2 isn't L_1^(L_2) !
```
cc: @d-torrance do you think this makes sense?

**Comment by d-torrance** [comment]
Yes, I think it would make sense if `_` and friends had higher precedence than  `^` and friends.  Subscripting having higher precedence than exponentiation seems to be common in other languages -- I checked R, Python, and Javascript.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Closing as `wontfix`, on a ground the thread never got to: the change would break `dd^C_i`.

That idiom — the `i`-th differential of a complex — works *because* `^` binds tighter than `_`, so
`dd^C_1` parses as `(dd^C)_1`. Under the proposed precedence it would parse as `dd^(C_1)`. It
appears **323 times across 23 files** in the distributed packages and Core, with another 95
occurrences in the `Complexes` documentation, and it turns up in essentially every piece of
workshop and textbook material that touches complexes.

@d-torrance's point above is sound in isolation — R, Python and JavaScript do all subscript before
exponentiating — but M2 has built a very widely used notation on the current order, and changing it
would silently alter the meaning of existing code rather than erroring on it.

Worth recording that the current behaviour does *not* silently compute the wrong thing. On 1.26.06:

```m2
i9 : y_1^y_2

stdio:8:3:(3):[1]: error: no method for binary operator ^ applied to objects:
            y  (of class S)
             1
      ^     y (of class IndexedVariableTable)
```

so `y_1^y_2` fails loudly rather than returning something unintended.

The underlying trap is real and closing this does not make it go away — the comment in
`HyperplaneArrangements.m2` that prompted the issue is still accurate, and anyone writing `L_1^L_2`
will still be surprised. If that is worth addressing, it would have to be by some route other than
a global precedence change.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
